src/main/java/core/basesyntax/Pair.java

```diff
@@ -1,5 +1,44 @@
 package core.basesyntax;
 
-class Pair {
+class Pair<F,S> {
+    private F first;
+    private S second;
 
+    public Pair(F first, S second) {
+        this.first = first;
+        this.second = second;
+    }
+
+    public F getFirst() {
+        return first;
+    }
+
+    public S getSecond() {
+        return second;
+    }
+
+    static <F, S> Pair of(F first, S second) {
+        return new Pair<>(first, second);
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        Pair<F,S> pair = (Pair) o;
+        return (first == null ? pair.first == null : first.equals(pair.first))
```

Comment:
try to make this check without ternary operator "?" and use only logical && and ||